internal/service/organization/resource_organization.go

```diff
@@ -74,7 +93,30 @@ func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.Resou
 			return nil
 		}
 
-		return diag.FromErr(fmt.Errorf("error create Organization: %s", err))
+		return diag.FromErr(fmt.Errorf("error creating Organization: %s", err))
+	}
+
+	orgID := *organization.Organization.Id
+
+	// update settings using new keys for this created organization because
+	// the provider/requesting API keys are not applicable for performing updates/delete for this new organization
+	cfg := config.Config{
+		PublicKey:  *organization.ApiKey.PublicKey,
+		PrivateKey: *organization.ApiKey.PrivateKey,
+		BaseURL:    meta.(*config.MongoDBClient).Config.BaseURL,
+	}
+
+	clients, _ := cfg.NewClient(ctx)
+	conn = clients.(*config.MongoDBClient).AtlasV2
+
+	_, _, err = conn.OrganizationsApi.UpdateOrganizationSettings(ctx, orgID, newOrganizationSettings(d)).Execute()
+	if err != nil {
+		if _, _, err := conn.OrganizationsApi.DeleteOrganization(ctx, orgID).Execute(); err != nil {
+			d.SetId("")
+			return diag.FromErr(fmt.Errorf("an error occurred when updating Organization settings. Unable to delete organization, there may be dangling resources: %s", err))
```

Comment:
is it possible here to return both errors?

Reply:
updated